connector/src/it/scala/com/datastax/bdp/spark/search/SearchAnalyticsIntegrationSpec.scala

```diff
@@ -96,12 +98,13 @@ class SearchAnalyticsIntegrationSpec extends SparkCassandraITFlatSpecBase with D
       )
       val ps = session.prepare(s"INSERT INTO $ks.$table (key, a, b, c) VALUES (?,?,?,?)")
       val groupedRows = generatedRows.grouped(100)
-      for (group <- groupedRows) {
-        val futures = for (row <- group) yield {
-          session.executeAsync(ps.bind(row.getInt(0): JInt, row.getInt(1): JInt, row.getInt(2): JInt, row.getString(3)))
+      awaitAll(
+        for (group <- groupedRows; row <- group) yield {
+          executor.executeAsync(ps.bind(row.getInt(0): JInt, row.getInt(1): JInt, row.getInt(2): JInt, row.getString(3)))
         }
-        futures.map(_.toCompletableFuture.get())
-      }
+      )
+
+      executor.waitForCurrentlyExecutingTasks()
```

Comment:
Given the refactor above do we still need this wait?  It doesn't hurt anything to keep it in since in anything but a truly aberrant case it'll be an immediate return.

Reply:
It doesn't, I mentioned in the Commit I was just being paranoid. I can remove them though if you like

Reply:
Nah, they're certainly not hurting anything and it's probably a worthwhile safeguard.  I'd say let's just leave 'em there.